From 5b3f5477e0a8b4a173e92206c22a8b43640f80eb Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Tue, 3 Sep 2013 21:07:57 +0300
Subject: [PATCH] aircrack-ptw-lib.c: get rid of textrel in rc4test_amd64_sse2

Original code hardcoded absolute references in asm file (x0123):

    movdqa (x0123), %%xmm0       \n\t

We force gcc to resolve it for us [1]

[1]: https://wiki.gentoo.org/wiki/Project:Hardened/HOWTO_locate_and_fix_textrels
     "How to make gcc worry about it" section:

Gentoo-bug: http://bugs.gentoo.org/483504
Reported-by: Rick Farina

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 src/aircrack-ptw-lib.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/aircrack-ptw-lib.c b/src/aircrack-ptw-lib.c
index 89a42fc..b7244b5 100644
--- a/src/aircrack-ptw-lib.c
+++ b/src/aircrack-ptw-lib.c
@@ -176,8 +176,8 @@ static int rc4test_amd64_sse2(uint8_t *key, int keylen, uint8_t *iv, uint8_t *ke
 		// key = keybuf
 		"lea  "keybuf", %q3           \n\t"
 		// load xmm registers
-		"movdqa (x0123), %%xmm0       \n\t"
-		"movdqa (x4444), %%xmm1       \n\t"
+		"movdqa %q9, %%xmm0       \n\t"
+		"movdqa %q10, %%xmm1       \n\t"
 		// clear some registers
 		"xorq      %q0, %q0           \n\t" // idx
 		"xorq      %q1, %q1           \n\t" // i
@@ -257,7 +257,7 @@ static int rc4test_amd64_sse2(uint8_t *key, int keylen, uint8_t *iv, uint8_t *ke
 	: "=&r"(idx), "=&r"(i), "=&r"(j),
 	  "+r"(key), "+r"(keylen), "+r"(iv), "+r"(keystream),
 	  "=&r"(scratch1), "=&r"(scratch2)
-	:
+	: "m"(x0123), "m"(x4444)
 	: "xmm0", "xmm1"
 	);
 #undef state
-- 
1.8.3.2

